source/registry/contracts/Registry.sol

```diff
@@ -65,13 +69,14 @@ contract Registry is Ownable {
     EnumerableSet.AddressSet storage supportedTokenList =
       supportedTokens[msg.sender];
     uint256 length = supportedTokenList.length();
-    for (uint256 i = 0; i < length; i++) {
-      address token = supportedTokenList.at(0);
+    uint256 originalLength = length;
+    while (length --> 0) {
```

Comment:
Prefer `length-- > 0` for clarity.

Reply:
worth noting: this behavior does not work in solidity 0.8 because it does not appreciate the underflow.

Reply:
of course that can be worked around with `unchecked`